packages/widget_toolkit_otp/lib/src/lib_sms_code_verification/bloc/sms_code_bloc.dart

```diff
@@ -0,0 +1,191 @@
+  SmsCodeBloc(
+      {required SmsCodeService service,
+      required CountdownService countdownService,
+      String? initialPhoneNumber,
+      this.sentNewCodeActivationTime = 3})
+      : _service = service,
+        _countDownService = countdownService,
+        _initialPhoneNumber = initialPhoneNumber {
+    // print('--SmsCodeBloc created');
+    if (_initialPhoneNumber != null) {
+      _number.add(_initialPhoneNumber!);
+    } else {
+      _$getPhoneNumberEvent
+          .startWith(null)
+          .switchMap((_) => _service.getFullPhoneNumber().asResultStream())
+          .setResultStateHandler(this)
+          .whereSuccess()
+          .bind(_number)
+          .addTo(_compositeSubscription);
+    }
+  }
+
+  final SmsCodeService _service;
+  final CountdownService _countDownService;
+  final String? _initialPhoneNumber;
+  final int sentNewCodeActivationTime;
+
+  final BehaviorSubject<String> _number = BehaviorSubject<String>();
+
+  @override
+  Stream<String> _mapToPhoneNumberState() => _number;
+
+  @override
+  Stream<ErrorModel?> _mapToErrorsState() => errorState.mapToErrorModel();
+
+  @override
+  Stream<Result<bool>> _mapToIsSendNewCodeEnabledState() => Rx.merge([
+        resendButtonThrottleTime.map((event) => event <= 0).asResultStream(),
+        _$sendNewCodeEvent
+            .withLatestFrom<String, String>(phoneNumber, (t, s) => s)
+            .switchMap(
+                (value) => _service.sendConfirmationSms(value).asResultStream())
+            .resetAfterCodeSent(this),
+        _$updatePhoneNumberEvent
+            .switchMap((value) => _service.updatePhoneNumber(value).asStream())
+            .doOnData((event) => _number.add(event))
+            .switchMap(
+                (value) => _service.sendConfirmationSms(value).asResultStream())
+            .resetAfterCodeSent(this),
+        onCodeVerificationResult
+            .where((event) => event == TemporaryCodeState.correct)
+            .mapTo(false)
+            .asResultStream()
+      ]);
+
+  @override
+  Stream<int> _mapToValidityTimeState() => _$resetEvent
+      .startWith(false)
+      .switchMap((value) => _service.getValidityTime(value).asStream())
+      .switchMap(
+        (time) => _countDownService
+            .countDown(maxTime: time)
+            .withLatestFrom(onCodeVerificationResult, (validity, state) {
+          final bool codeNotSent = state != TemporaryCodeState.correct &&
+              state != TemporaryCodeState.wrong;
+          if (validity == 0 && codeNotSent) {
+            setTemporaryCodeState(TemporaryCodeState.disabled);
+          }
+          return validity;
+        }),
+      );
+
+  @override
+  Stream<bool> _mapToIsLoadingState() => loadingState;
+
+  @override
+  Stream<TemporaryCodeState> _mapToOnCodeVerificationResultState() => Rx.merge([
+        _$verifyCodeEvent.flatMap((value) => confirmSMSCode(value)
+            .asResultStream()
+            .setResultStateHandler(this)
```

Comment:
The bloc must registered as state handler just once, not every time when _verifyCodeEvent is called